scripts/gather-mocked-data.js

```diff
@@ -0,0 +1,186 @@
+    let fullUrl = `${serverAddr}/${url}`;
+    return client
+      .get(fullUrl)
+      .then(response => {
+        return { ...v, data: response.data, url: url, serverAddr: serverAddr };
+      })
+      .catch(error => {
+        console.log(error);
+        process.exit(1);
+      });
+  });
+  //  Reducing to a single promise that has collected all of the results.
+  //  Transform from array to a dict that matches the layout of how we want to store the data
+  return all.reduce((promiseChain, currentTask) => {
+    return promiseChain.then(chainResults => {
+      return currentTask.then(currentResult => {
+        chainResults[currentResult.url] = currentResult;
+        return chainResults;
+      });
+    });
+  }, Promise.resolve({}));
+}
+
+function collectAndReformat(data) {
+  // Reformatting data into format of
+  //  { 'url':  response_data }
+  return data.then(results => {
+    return Object.keys(results).reduce(function(acc, key) {
+      let entry = results[key].data.items.reduce((acc, current) => {
+        // Example of 'current'
+        //  {apiVersion: 'migration.openshift.io/v1alpha1',
+        //  kind: 'MigStorage',
+        //  metadata: [Object],
+        //  spec: [Object]}
+        acc[current.metadata.name] = current;
+        return acc;
+      }, {});
+      acc[key] = entry;
+      return acc;
+    }, {});
+  });
+}
+
+function gatherSecretRefs(client, serverAddr, data) {
+  let migstorages = data['apis/migration.openshift.io/v1alpha1/namespaces/mig/migstorages'];
+  let storageKeys = Object.keys(migstorages);
+  let storageSecretRefs = storageKeys.reduce((acc, currentKey) => {
+    return acc
+      .concat(migstorages[currentKey].spec.backupStorageConfig.credsSecretRef)
+      .concat(migstorages[currentKey].spec.volumeSnapshotConfig.credsSecretRef);
+  }, []);
+
+  let migclusters = data['apis/migration.openshift.io/v1alpha1/namespaces/mig/migclusters'];
+  let clusterKeys = Object.keys(migclusters);
+  let clusterSecretRefs = clusterKeys.reduce((acc, currentKey) => {
+    return acc.concat(migclusters[currentKey].spec.serviceAccountSecretRef);
+  }, []);
+  let secretRefs = [].concat(storageSecretRefs).concat(clusterSecretRefs);
+
+  let all = secretRefs.map(s => {
+    let key = `api/v1/namespaces/${s.namespace}/secrets`;
+    let fullUrl = `${serverAddr}/${key}/${s.name}`;
+    return client
+      .get(fullUrl)
+      .then(response => {
+        return { ...s, key: key, data: response.data, serverAddr: serverAddr };
+      })
+      .catch(error => {
+        console.log(error);
+        process.exit(1);
+      });
+  });
+
+  return all.reduce((promiseChain, currentTask) => {
+    return promiseChain.then(chainResults => {
+      return currentTask.then(currentResult => {
+        if (!chainResults[currentResult.key]) {
+          chainResults[currentResult.key] = {};
+        }
+        chainResults[currentResult.key][currentResult.name] = currentResult.data;
```

Comment:
This data needs to be sanitized for any aws access keys to ensure we don't accidentally commit real data.

Reply:
Resolved